pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -53,20 +53,18 @@ type ClientCache struct {
 
 	matchVersion bool
 
-	defaultConfigLock sync.Mutex
-	defaultConfig     *restclient.Config
+	mutex         sync.Mutex
```

Comment:
Or rather - if there is one lock in a struct, use `lock`.  If there are multiple, call them `stuffLock` relative to the things they lock.  Also, generally avoid multiple locks anyway because that way lies madness.